### PR TITLE
JAMES-2948 Clarify and fix user creation

### PR DESCRIPTION
**Pull request by blackheaven**

### CODE REVIEW

File: server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/UsersRoutesTest.java [hunk 1]
```diff
@@ -173,7 +173,7 @@ void postShouldAddTheUser() {
         }
 
         @Test
-        void postingTwoTimesShouldBeAllowed() {
+        void putingTwoTimesShouldBeAllowed() {
```
Comment (rouazana) [inline server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/UsersRoutesTest.java]: putting

File: server/protocols/webadmin/webadmin-data/pom.xml [hunk 2]
```diff
@@ -87,6 +87,12 @@
             <groupId>com.google.guava</groupId>
             <artifactId>guava</artifactId>
         </dependency>
+        <dependency>
+            <groupId>commons-lang</groupId>
+            <artifactId>commons-lang</artifactId>
```
Comment (trantienduchn) [inline server/protocols/webadmin/webadmin-data/pom.xml]: you can use `commons-lang3` which is already declared in the root pom. Or maybe you don't need it at all.

File: server/protocols/webadmin/webadmin-data/pom.xml [hunk 3]
```diff
@@ -88,9 +88,12 @@
...
-            <groupId>commons-lang</groupId>
-            <artifactId>commons-lang</artifactId>
-            <version>2.6</version>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-lang3</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-lang3</artifactId>
             <scope>test</scope>
```
Comment (chibenwa) [inline server/protocols/webadmin/webadmin-data/pom.xml]: This import is duplicated with the previous one?

File: server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/service/UserService.java [hunk 4]
```diff
@@ -88,8 +85,7 @@ public String upsertUser(String username, char[] password, Response response) th
 
     private void usernamePreconditions(String username) {
         try {
-            Preconditions.checkArgument(!Strings.isNullOrEmpty(username));
-            Preconditions.checkArgument(username.length() <= MAXIMUM_MAIL_ADDRESS_LENGTH);
+            org.apache.james.core.User.fromUsername(username);
```
Comment (Arsnael) [inline server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/service/UserService.java]: Import `User` class?

**Comment by Arsnael** [comment]
CI restart...

test this please

File: server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/service/UserService.java (hunk 4 above)
Comment (blackheaven) [reply server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/service/UserService.java]: Sadly, there is already another `User` class imported

Comment (chibenwa) [reply server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/service/UserService.java]: @rouazana work should fix this.

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**